packages/dev/core/src/Cameras/arcRotateCamera.ts

```diff
@@ -914,7 +914,7 @@ export class ArcRotateCamera extends TargetCamera {
             const handednessMultiplier = this._calculateHandednessMultiplier();
             let inertialAlphaOffset = this.inertialAlphaOffset * handednessMultiplier;
 
-            if (this.beta <= 0) {
+            if (this.beta < 0 || (this.allowUpsideDown && this.beta === 0)) {

```

Comment:
So, here's what I'm seeing.  With respect to the provided PG, the issue appears to me that when `beta = 0`, the camera will move in the opposite direction and then will switch back to the right direction when `beta` increases above 0 again.  With your fixed, `allowUpsideDown` being `false` is effectively negating the issue and reducing the inversion check to `if (this.beta < 0)`.  In this case, it might be better to just go with `if (this.beta < 0)`.